### PR TITLE
feat(): expose host flavor

### PR DESCRIPTION
**Pull request by vician**
### Description

Exposing host flavor variable the exact way as https://github.com/terraform-ibm-modules/terraform-ibm-icd-postgresql/pull/382

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [X] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Added support for host flavor variable.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -82,20 +85,39 @@ resource "ibm_database" "redis_database" {
...
-    memory {
-      allocation_mb = var.memory_mb
-    }
-    disk {
-      allocation_mb = var.disk_mb
-    }
-    cpu {
-      allocation_count = var.cpu_count
+  dynamic "group" {
+    for_each = local.host_flavor_set ? [1] : []
```
Comment (shemau) [inline main.tf:89]: Based on postgres, this is insufficient (for all use cases). I assume redis supports (member)_host_flavor="multitenant". reference terraform-ibm-icd-postgres for implementation details. Also consider adding multitenant support via other_test.go.

File: variables.tf [hunk 2]
```diff
@@ -233,3 +233,9 @@ variable "backup_crn" {
     error_message = "backup_crn must be null OR starts with 'crn:' and contains ':backup:'"
   }
 }
+
+variable "member_host_flavor" {
+  type        = string
+  description = "Allocated host flavor per member. For more information, see https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/database#host_flavor"
+  default     = null
+}
```
Comment (shemau) [inline variables.tf:241]: When adding variables to the main module also add them to sub modules, especially fscloud (redis has this, so it needs updating) and any DAs (redis does not have one of these).
postgres also shows how to add this to fscloud sub module.

**Review by shemau — CHANGES_REQUESTED** [review]
This looks good, but could be more consistent with postgres. Specially support for `multitenant` and extension into fscloud sub module. See comments and reference postgres.

File: main.tf (hunk 1 above)
Comment (vician) [reply main.tf]: Thank you @shemau I addressed that

File: main.tf [hunk 3]
```diff
@@ -83,19 +86,42 @@ resource "ibm_database" "redis_database" {
...
+        allocation_mb = var.memory_mb
+      }
+      disk {
+        allocation_mb = var.disk_mb
+      }
+      cpu {
+        allocation_count = var.cpu_count
+      }
+      members {
+        allocation_count = var.members
```
Comment (shemau) [inline main.tf:123]: This block should have a host_flavor entry (which will "multi tenant")

```hcl
     host_flavor {
        id = var.member_host_flavor
      }
```

**Review by shemau — CHANGES_REQUESTED** [review]

**Comment by shemau** [comment]
This will now be addressed by #368 which part of an epic rolling support across all ICD modules

**Pull request closed without merging**